### ISSUE TITLE
Cutter error handling

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#412]
In issue #401 I said I'd open a new issue for this, but apparently I never did. Here are comments from Evan that need to be followed up:

+ The only time a dialog informing them of no documents comes up when you're in cut by milestone and have entered a string to cut by. In the other options, the default cutting error modal pops up whenever there is nothing entered in that input, and when there is something there, the invalid overlap error appears and the user will never actually receive the "no documents" message.

+ The only error I could not get is err6 "Individual cutting: Invalid Segment size", which gets a "Lexos could not apply scrubbing actions".

**Comment by scottkleinman** [comment]
The reason the invalid overlap error keeps popping up and the no active documents error never gets displayed is that the script was looping through the errors array and displaying each error in turn, with the overlap error being the last item in the array. Apparently, you can display Bootstrap modals in succession in this way (at least, not without some provision for interrupting the loop to get a user response). So, for now, I've eliminated the loop and displayed the first error. Once that is fixed, the second error will move to position 1 and will be displayed. That should work until such a time as we feel like we need to get more sophisticated.

**Comment by scottkleinman** [comment]
```javascript
// Make sure the individual segment size is valid       
if ((Math.abs(Math.round(individualCutValue)) != individualCutValue)) {
    errors.push(err6);
}
```
It looks to me like this is checking to make sure that the individual cut value is a non-negative integer. Error 4 does a similar thing:

```javascript
// Make sure the overall segment size is valid      
if((Math.abs(Math.round(overallcutvalue)) != overallcutvalue) || overallcutvalue == 0) {
    errors.push(err4);
}
```
But, if you enter "3.5" as the cut value, you get the generic error. This is because the data is being passed to the server and failing on the back end. In other words, the if conditions are not catching error. It seems to work when the value is "0", but not when it is a decimal of null. Feel free to chime in if you know why.

**Comment by evanlaferriere** [comment]
I fixed the no documents error not showing up yesterday, by making it an if/else for whether or not there were any files instead of just an if

**Comment by evanlaferriere** [comment]
I was successfully getting the error for decimals thrown the other day, I thought. I definitely did get err4, I can look into it. 
A pattern I've noticed for when cutter breaks/doCutting not being called is that it tends to happen after returning to the page after a while, and seemingly never after I've successfully used the tool recently. Which seems to fit with the caching causing that error

**Comment by evanlaferriere** [comment]
I'm pushing a little piece of code that check to see if the loading icon has been going for a while and says "It seems to be taking a while to load. If you're aren't processing a large number of documents, please reload the page and try again." I have it set to go off after ten seconds of loading.

**Comment by scottkleinman** [comment]
Your code pops up in the error-modal, which looks a little threatening, but there's no easy way to fix that without disrupting error messages on other pages. I suggest we leave that for another time.

Meanwhile, I've pushed several fixes which should take care of the remaining issues. I'll close this issue. We can re-open it if we need to.

**Comment by scottkleinman** [comment]
I apparently missed this report:

+ Just noticed that I'm not getting the "Over 100 new segments" warning when I'm cutting thousands of docs and definitely should be gettting that message.

I'd better double check this before closing this issue.

**Comment by evanlaferriere** [comment]
Yeah, I haven't been able to produce it since I started testing cutter a few days ago so I think it's been taken out by accident

**Comment by evanlaferriere** [comment]
Oh nvm I just got it, doing a big ol' cut
I think we're done here

**Comment by evanlaferriere** [comment]
nvm when I get that error it stops the whole process

**Comment by evanlaferriere** [comment]
```
scripts_cut.js?ver=3.0:29 Uncaught ReferenceError: xhr is not defined
```
whenever I get the modal fr 100 new segments this gets thrown and it doesn't complete

**Comment by evanlaferriere** [comment]
Something on upload is throwing up an xhr message too
```
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
```
What is xhr being used for?

**Comment by evanlaferriere** [comment]
So cutter is fast enough that it can make 100 cuts pretty easy. It's working on lots of files that slows it down. We should take out that 100 new segment modal because when it pops up it cancels the cutting action regardless of whether you click continue or click. The error I made yesterday that triggers during a long load covers when a large number of files is actually an issue, and does not stop anything, I think it's already a better alternative to what's going on now.

**Comment by scottkleinman** [comment]
Merged into #447.
